e3sm_diags/driver/lat_lon_driver.py

```diff
@@ -2,242 +2,427 @@
+        a sub-dictionary (key is metric and value is float) or a string
+        ("unit").
+    """
+    test_var = ds_test[var_key]
+    test_regrid_var = ds_test_regrid[var_key]
+
+    # xarray.DataArray.min() and max() returns a `np.ndarray` with a single
+    # int/float element. Using `.item()` returns that single element.
+    metrics_dict = {
+        "test": {
+            "min": test_var.min().item(),
+            "max": test_regrid_var.max().item(),
+            "mean": spatial_avg(ds_test, var_key, serialize=True),
+        },
+        "test_regrid": {
+            "min": test_regrid_var.min().item(),
+            "max": test_regrid_var.max().item(),
+            "mean": spatial_avg(ds_test_regrid, var_key, serialize=True),
+            "std": std(ds_test_regrid, var_key, serialize=True),
+        },
+        "ref": {
+            "min": None,
+            "max": None,
+            "mean": None,
+        },
+        "ref_regrid": {
+            "min": None,
+            "max": None,
+            "mean": None,
+            "std": None,
+        },
+        "misc": {
+            "rmse": None,
+            "corr": None,
+        },
+        "diff": {
+            "min": None,
+            "max": None,
+            "mean": None,
+        },
+        "unit": ds_test[var_key].attrs["units"],
+    }
 
-            else:
-                raise RuntimeError(
-                    "Dimensions of the two variables are different. Aborting."
-                )
+    if ds_ref is not None:
+        ref_var = ds_ref[var_key]
+
+        metrics_dict["ref"] = {
+            "min": ref_var.min().item(),
+            "max": ref_var.max().item(),
+            "mean": spatial_avg(ds_ref, var_key, serialize=True),
+        }
+
+    if ds_ref_regrid is not None:
+        ref_regrid_var = ds_ref_regrid[var_key]
+
+        metrics_dict["ref_regrid"] = {
+            "min": ref_regrid_var.min().item(),
+            "max": ref_regrid_var.max().item(),
+            "mean": spatial_avg(ds_ref_regrid, var_key, serialize=True),
+            "std": std(ds_ref_regrid, var_key, serialize=True),
+        }
+
+        weights = get_weights(ds_test_regrid)
+        metrics_dict["misc"] = {
+            "rmse": rmse(
+                test_regrid_var,
+                ref_regrid_var,
+                weights=weights,
+                serialize=True,
+            ),
+            "corr": correlation(
+                test_regrid_var,
+                ref_regrid_var,
+                weights=weights,
+                serialize=True,
+            ),
+        }
+
+    if ds_diff is not None:
+        diff_var = ds_diff[var_key]
+
+        metrics_dict["diff"] = {
+            "min": diff_var.min().item(),
+            "max": diff_var.max().item(),
+            "mean": spatial_avg(ds_diff, var_key, serialize=True),
+        }
 
-    return parameter
+    return metrics_dict
+
+
+def _save_and_plot_metrics(
```

Comment:
how about to change to `_save_and_plot_data_metrics`?

Reply:
Sounds good, I'll update the function name

Reply:
Actually, I'll update it to `_save_and_plot_metrics_dict()` because I call the dictionary variable `metrics_dict()` and the function `_create_metrics_dict()`.

Reply:
in the plotter, both data and metrics are used, only only metrics. So maybe stick `data` in there?

Reply:
Oh I think I see what you're saying. We're saving and plotting data _and_ metrics, and not appending "data" to "metrics". That makes sense to me, I'll make the change.